Automattic/wp-calypso
# Stats: Replace moment with luxon in StatsChartTabs

#### Proposed Changes

* Adds `luxon` npm dependency to the `stats` bundle.
* Replaces `moment` invocation with `luxon` in the StatsChartTabs component.
* Migrates refactored functions to a TypeScript module.
* Adds unit tests.

#### Testing Instructions



* Open this PR in a live branch.
* Navigate to the Stats section (`/stats/day/:siteSlug`)
* Try changing the period types (e.g., "weeks", "months", "years"). Ensure that they work as expected.
* Use the arrows to move between dates. Ensure that the chart tabs update as expected.
* Try going beyond the boundary of what's currently being shown by pressing the left arrow button while having the first bar selected. Note that this is only possible with sites with a sizable stats history. Ensure that a new set of 30 vertical bars (or 10 vertical bars for the "years" display) renders as expected.

Also, ensure that the new unit tests pass as expected.

#### Pre-merge Checklist





Part of #70720.
